draft-ietf-dhc-addr-notification.md

```diff
@@ -322,22 +322,24 @@ If an ADDR-REG-REPLY message is received for the address being registered, the c
+We define a function AddrRegRefreshInterval(address) as min(4 hours, 80% of the address's current Valid Lifetime). When calculating this value, the client SHOULD apply a jitter of +/- 10% to avoid synchronization causing a large number of registration messages from different clients at the same time.
 
-1. If no refresh is currently scheduled, it MUST register immediately and schedule a refresh.
-1. If a refresh is currently scheduled, it MUST reschedule the existing refresh if this would result in the refresh being sooner than currently scheduled.
+Whenever the client registers or refreshes an address, it calculates a NextAddrRegRefreshTime for that address as AddrRegRefreshInterval seconds in the future, but does not schedule any refreshes.
+
+Whenever the client receives a PIO which changes the Valid Lifetime of an existing address by more than 1%, then the client calculates a new AddrRegRefreshInterval. The client schedules a refresh for min(now + AddrRegRefreshInterval, NextAddrRegRefreshTime). If the refresh would be scheduled in the past, then the refresh occurs immediately.
 
 When a refresh is performed, the client MAY refresh all addresses assigned to the interface that are scheduled to be refreshed within the next AddrRegRefreshCoalesce seconds. The value of AddrRegRefreshCoalesce is implementation-dependent, and a suggested default is 60 seconds.
 
-Discussion: this algorithm ensures that refreshes are not sent too frequently, while ensuring that the server never believes that the address has expired when it has not. Specifically:
+Discussion: this algorithm ensures that refreshes are not sent too frequently, while ensuring that the server never believes that the address has expired when it has not. Specifically, after every registration:
 
-- If the network never changes the lifetime, or stops refreshing the lifetime, then only one refresh ever occurs. The address expires.
-- Point #1 ensures that any time the network changes the lifetime when no refresh is scheduled, the server will be informed of the correct lifetime. If the network does not change the address's lifetime, then the server already knows the correct lifetime and no refresh needs to be sent.
-- Point #2 ensures that if the network reduces the lifetime of the address, then the server will be informed of the new lifetime. If the network increases the lifetime of the address, the refresh will be sent at the previously scheduled time, and the server will be informed of the correct lifetime. From this point on, either the address expires (and the server is informed of when this will happen) or an RA increases the lifetime, in which case a refresh will be sent.
+- If the network never changes the lifetime, or stops refreshing the lifetime, then no refreshes occur. Refreshes are not necessary, because the address expires at the time the server expects it to expire.
```

Comment:
s/If the network never changes the lifetime, or stops refreshing the lifetime/If the client stops receiving RAs refreshing the lifetime/ maybe..